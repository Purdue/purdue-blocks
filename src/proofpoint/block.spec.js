// title-nav test
jest.setTimeout(30000)
import {
    createNewPost,
    enablePageDialogAccept,
    getEditedPostContent,
    insertBlock,
} from '@wordpress/e2e-test-utils';

import {
    blockStartup,
    clickCheckbox,
    clickRadio,
} from '../test-helpers'

const block = {blockTitle: 'Reasons to Believe', blockName: 'purdue-blocks/proofpoint'}

describe( 'ðŸ”¬ Proofpoint Block', () => {
    beforeAll( async () => {
        await enablePageDialogAccept();
    } );
    beforeEach( async () => {
        await createNewPost();
    } );

    test( 'Block should be available.', async () => {
        await insertBlock( block.blockTitle )


        // tests that the block is properly inserted and matches the existing snapshot
        expect(await page.$(`[data-type="${block.blockName}"]`)).not.toBeNull()
        expect( await getEditedPostContent() ).toMatchSnapshot()
    } )

    describe('ðŸ”¬ Block Editor Fields', () => {
        describe('ðŸ”¬ Proofpoint body text fields', () => {
            test('ðŸ”Ž Highlighted Text field updates output correctly', async () => {
                await blockStartup(block)

                const typeString = "Highlighted text test."
    
                await page.focus('input[placeholder="Highlighted Text..."]')
                await page.keyboard.type(typeString, {delay: 10})
    
                const editedContent = await getEditedPostContent()
                expect( editedContent.includes(`"highlighted":"${typeString}"`)).toBe(true)
                expect( await getEditedPostContent() ).toMatchSnapshot()
            })
            test('ðŸ”Ž Content Text field updates output correctly', async () => {
                await blockStartup(block)

                const typeString = "Content text test."
    
                await page.focus('input[placeholder="Content Text..."]')
                await page.keyboard.type(typeString, {delay: 10})
    
                const editedContent = await getEditedPostContent()
                expect( editedContent.includes(`"content":"${typeString}"`)).toBe(true)
                expect( await getEditedPostContent() ).toMatchSnapshot()
            })
            test('ðŸ”Ž Proofpoint source field updates output correctly', async () => {
                await blockStartup(block)

                const typeString = "Proofpoint source test."
    
                await page.focus('input[placeholder="Source of Text..."]')
                await page.keyboard.type(typeString, {delay: 10})
    
                const editedContent = await getEditedPostContent()
                expect( editedContent.includes(`"source":"${typeString}"`)).toBe(true)
                expect( await getEditedPostContent() ).toMatchSnapshot()
            })
        })

        describe('ðŸ”¬ Proofpoint CTA fields', () => {
            test('ðŸ”Ž CTA text field updates output correctly', async () => {
                await blockStartup(block)

                const typeString = "CTA text"
    
                await page.focus('input[placeholder="CTA Text..."]')
                await page.keyboard.type(typeString, {delay: 10})
    
                const editedContent = await getEditedPostContent()
                expect( editedContent.includes(`"ctaText":"${typeString}"`)).toBe(true)
                expect( await getEditedPostContent() ).toMatchSnapshot()
            })
            test('ðŸ”Ž CTA url field updates output correctly', async () => {
                await blockStartup(block)

                const typeString = "https://www.purdue.edu"
    
                await page.focus('input[placeholder="CTA URL..."]')
                await page.keyboard.type(typeString, {delay: 10})
    
                const editedContent = await getEditedPostContent()
                expect( editedContent.includes(`"ctaUrl":"${typeString}"`)).toBe(true)
                expect( await getEditedPostContent() ).toMatchSnapshot()
            })
        })        
    })

    describe('ðŸ”¬ Side Panel Settings', () => {
        describe('ðŸ”¬ Background color controls', () => {
            test('ðŸ”Ž White background option reveals add border checkbox', async () => {
                await blockStartup(block)

                await clickRadio('Background Color', 'white')

                const [borderCheckbox] = await page.$x( `//label[@class="components-checkbox-control__label"][contains(text(),"Add border?")]` )

                expect( borderCheckbox ).not.toBeNull()
                expect( borderCheckbox ).not.toBe(undefined)
                expect( await getEditedPostContent() ).toMatchSnapshot()
            })

            test('ðŸ”Ž Add border checkbox correctly adds borders', async () => {
                await blockStartup(block)

                await clickRadio('Background Color', 'white')
                await clickCheckbox('Add border?')
                
                const editedContent = await getEditedPostContent()
                expect( editedContent.includes(`"border":true`)).toBe(true)
                expect( await getEditedPostContent() ).toMatchSnapshot()
            })

            test('ðŸ”Ž Button color on white background updates output correctly', async () => {
                await blockStartup(block)

                await clickRadio('Background Color', 'white')
                await clickRadio('Button Color', 'white')

                let editedContent = await getEditedPostContent()
                expect( editedContent.includes(`"buttonColor":"white"`)).toBe(true)
                expect( await getEditedPostContent() ).toMatchSnapshot()


                await clickRadio('Button Color', 'black')

                editedContent = await getEditedPostContent()
                expect( editedContent.includes(`"buttonColor":"white"`)).toBe(false)
                expect( await getEditedPostContent() ).toMatchSnapshot()
            })
        })

        describe('ðŸ”¬ Text Style Settings', () => {
            test('ðŸ”Ž Highlighted text style radio controls', async () => {
                await blockStartup(block)

                // type in some content so that the p el is added to output to test text style setting
                const typeString = "Highlighted text test."
                await page.focus('input[placeholder="Highlighted Text..."]')
                await page.keyboard.type(typeString, {delay: 10})

                const [wideControl] = await page.$x( `//div[@class="components-radio-control__option"][../label='Highlighted Text Style']//input[@value='wide']` )
                await wideControl.click()
                
                let editedContent = await getEditedPostContent()
                expect( editedContent.includes(`"headerfontStyle":"wide"`)).toBe(true)
                expect( await getEditedPostContent() ).toMatchSnapshot()

                // then test that it swaps back correctly
                const [narrowControl] = await page.$x( `//div[@class="components-radio-control__option"][../label='Highlighted Text Style']//input[@value='narrow']` )
                await narrowControl.click()
                
                editedContent = await getEditedPostContent()
                expect( editedContent.includes(`"headerfontStyle":"wide"`)).toBe(false)
                expect( await getEditedPostContent() ).toMatchSnapshot()
            })

            test('ðŸ”Ž Content text style radio controls', async () => {
                await blockStartup(block)

                // type in some content so that the p el is added to output to test text style setting
                const typeString = "Content text test."
                await page.focus('input[placeholder="Content Text..."]')
                await page.keyboard.type(typeString, {delay: 10})

                await clickRadio('Content Text Style', 'wide')
                
                let editedContent = await getEditedPostContent()
                expect( editedContent.includes(`"contentfontStyle":"wide"`)).toBe(true)
                expect( await getEditedPostContent() ).toMatchSnapshot()

                // then test that it swaps back correctly
                await clickRadio('Content Text Style', 'narrow')
                
                editedContent = await getEditedPostContent()
                expect( editedContent.includes(`"contentfontStyle":"wide"`)).toBe(false)
                expect( await getEditedPostContent() ).toMatchSnapshot()
            })
        })
        describe('ðŸ”¬ Button Link Setting', () => {
            test('ðŸ”Ž Link can be opened in new tab', async () => {
                await blockStartup(block)

                await clickCheckbox('Open link in new tab?')

                const editedContent = await getEditedPostContent()
                expect( editedContent.includes(`"external":true`)).toBe(true)
                expect( await getEditedPostContent() ).toMatchSnapshot()
            })
        })
    })
})