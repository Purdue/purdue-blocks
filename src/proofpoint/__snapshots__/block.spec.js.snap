// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`🔬 Proofpoint Block Block should be available. 1`] = `
"<!-- wp:purdue-blocks/proofpoint -->
<div class=\\"pu-proofpoint pu-proofpoint__black\\"><div class=\\"container\\"></div></div>
<!-- /wp:purdue-blocks/proofpoint -->"
`;

exports[`🔬 Proofpoint Block 🔬 Block Editor Fields 🔬 Proofpoint CTA fields 🔎 CTA text field updates output correctly 1`] = `
"<!-- wp:purdue-blocks/proofpoint {\\"ctaText\\":\\"CTA text\\"} -->
<div class=\\"pu-proofpoint pu-proofpoint__black\\"><div class=\\"container\\"></div></div>
<!-- /wp:purdue-blocks/proofpoint -->"
`;

exports[`🔬 Proofpoint Block 🔬 Block Editor Fields 🔬 Proofpoint CTA fields 🔎 CTA url field updates output correctly 1`] = `
"<!-- wp:purdue-blocks/proofpoint {\\"ctaUrl\\":\\"https://www.purdue.edu\\"} -->
<div class=\\"pu-proofpoint pu-proofpoint__black\\"><div class=\\"container\\"></div></div>
<!-- /wp:purdue-blocks/proofpoint -->"
`;

exports[`🔬 Proofpoint Block 🔬 Block Editor Fields 🔬 Proofpoint body text fields 🔎 Content Text field updates output correctly 1`] = `
"<!-- wp:purdue-blocks/proofpoint {\\"content\\":\\"Content text test.\\"} -->
<div class=\\"pu-proofpoint pu-proofpoint__black\\"><div class=\\"container\\"><p class=\\"pu-proofpoint__content pu-proofpoint__content-narrow\\">Content text test.</p></div></div>
<!-- /wp:purdue-blocks/proofpoint -->"
`;

exports[`🔬 Proofpoint Block 🔬 Block Editor Fields 🔬 Proofpoint body text fields 🔎 Highlighted Text field updates output correctly 1`] = `
"<!-- wp:purdue-blocks/proofpoint {\\"highlighted\\":\\"Highlighted text test.\\"} -->
<div class=\\"pu-proofpoint pu-proofpoint__black\\"><div class=\\"container\\"><p class=\\"pu-proofpoint__highlighted pu-proofpoint__highlighted-narrow\\">Highlighted text test.</p></div></div>
<!-- /wp:purdue-blocks/proofpoint -->"
`;

exports[`🔬 Proofpoint Block 🔬 Block Editor Fields 🔬 Proofpoint body text fields 🔎 Proofpoint source field updates output correctly 1`] = `
"<!-- wp:purdue-blocks/proofpoint {\\"source\\":\\"Proofpoint source test.\\"} -->
<div class=\\"pu-proofpoint pu-proofpoint__black\\"><div class=\\"container\\"><p class=\\"pu-proofpoint__source\\">Proofpoint source test.</p></div></div>
<!-- /wp:purdue-blocks/proofpoint -->"
`;

exports[`🔬 Proofpoint Block 🔬 Side Panel Settings 🔬 Background color controls 🔎 Add border checkbox correctly adds borders 1`] = `
"<!-- wp:purdue-blocks/proofpoint {\\"color\\":\\"white\\",\\"border\\":true} -->
<div class=\\"pu-proofpoint pu-proofpoint__white pu-proofpoint__border\\"><div class=\\"container\\"></div></div>
<!-- /wp:purdue-blocks/proofpoint -->"
`;

exports[`🔬 Proofpoint Block 🔬 Side Panel Settings 🔬 Background color controls 🔎 Button color on white background updates output correctly 1`] = `
"<!-- wp:purdue-blocks/proofpoint {\\"color\\":\\"white\\",\\"buttonColor\\":\\"white\\"} -->
<div class=\\"pu-proofpoint pu-proofpoint__white\\"><div class=\\"container\\"></div></div>
<!-- /wp:purdue-blocks/proofpoint -->"
`;

exports[`🔬 Proofpoint Block 🔬 Side Panel Settings 🔬 Background color controls 🔎 Button color on white background updates output correctly 2`] = `
"<!-- wp:purdue-blocks/proofpoint {\\"color\\":\\"white\\"} -->
<div class=\\"pu-proofpoint pu-proofpoint__white\\"><div class=\\"container\\"></div></div>
<!-- /wp:purdue-blocks/proofpoint -->"
`;

exports[`🔬 Proofpoint Block 🔬 Side Panel Settings 🔬 Background color controls 🔎 White background option reveals add border checkbox 1`] = `
"<!-- wp:purdue-blocks/proofpoint {\\"color\\":\\"white\\"} -->
<div class=\\"pu-proofpoint pu-proofpoint__white\\"><div class=\\"container\\"></div></div>
<!-- /wp:purdue-blocks/proofpoint -->"
`;

exports[`🔬 Proofpoint Block 🔬 Side Panel Settings 🔬 Text Style Settings 🔎 Content text style radio controls 1`] = `
"<!-- wp:purdue-blocks/proofpoint {\\"contentfontStyle\\":\\"wide\\",\\"content\\":\\"Content text test.\\"} -->
<div class=\\"pu-proofpoint pu-proofpoint__black\\"><div class=\\"container\\"><p class=\\"pu-proofpoint__content pu-proofpoint__content-wide\\">Content text test.</p></div></div>
<!-- /wp:purdue-blocks/proofpoint -->"
`;

exports[`🔬 Proofpoint Block 🔬 Side Panel Settings 🔬 Text Style Settings 🔎 Content text style radio controls 2`] = `
"<!-- wp:purdue-blocks/proofpoint {\\"content\\":\\"Content text test.\\"} -->
<div class=\\"pu-proofpoint pu-proofpoint__black\\"><div class=\\"container\\"><p class=\\"pu-proofpoint__content pu-proofpoint__content-narrow\\">Content text test.</p></div></div>
<!-- /wp:purdue-blocks/proofpoint -->"
`;

exports[`🔬 Proofpoint Block 🔬 Side Panel Settings 🔬 Text Style Settings 🔎 Highlighted text style radio controls 1`] = `
"<!-- wp:purdue-blocks/proofpoint {\\"highlighted\\":\\"Highlighted text test.\\",\\"headerfontStyle\\":\\"wide\\"} -->
<div class=\\"pu-proofpoint pu-proofpoint__black\\"><div class=\\"container\\"><p class=\\"pu-proofpoint__highlighted pu-proofpoint__highlighted-wide\\">Highlighted text test.</p></div></div>
<!-- /wp:purdue-blocks/proofpoint -->"
`;

exports[`🔬 Proofpoint Block 🔬 Side Panel Settings 🔬 Text Style Settings 🔎 Highlighted text style radio controls 2`] = `
"<!-- wp:purdue-blocks/proofpoint {\\"highlighted\\":\\"Highlighted text test.\\"} -->
<div class=\\"pu-proofpoint pu-proofpoint__black\\"><div class=\\"container\\"><p class=\\"pu-proofpoint__highlighted pu-proofpoint__highlighted-narrow\\">Highlighted text test.</p></div></div>
<!-- /wp:purdue-blocks/proofpoint -->"
`;
